internal/core/command/get_test.go

```diff
@@ -108,9 +116,99 @@ func TestNewGetCommandNoCorrelationIDInContext(t *testing.T) {
 
 func TestNewGetCommandInvalidBaseUrl(t *testing.T) {
 	device := testDevice
+	req := newRequestWithHeaders(map[string]string{clients.ContentType: clients.ContentTypeCBOR}, http.MethodGet)
 	device.Service.Addressable.Address = "!@#$"
-	_, err := NewGetCommand(device, testCommand, "", context.Background(), nil, logger.NewMockClient())
+	_, err := NewGetCommand(device, testCommand, context.Background(), nil, logger.NewMockClient(), req)
 	if err == nil {
 		t.Errorf("The invalid URL error was not properly propagated to the caller")
 	}
 }
+
+func TestNewGetCommandContentType(t *testing.T) {
+	tests := []struct {
+		name            string
+		expectedHeaders map[string]string
+		req             *http.Request
+	}{
+		{
+			"cbor content type header propagated",
+			map[string]string{
+				clients.ContentType: clients.ContentTypeCBOR},
+			newRequestWithHeaders(map[string]string{
+				clients.ContentType: clients.ContentTypeCBOR}, http.MethodGet),
+		},
+		{
+			"json content type header propagated",
+			map[string]string{
+				clients.ContentType: clients.ContentTypeJSON},
+			newRequestWithHeaders(map[string]string{
+				clients.ContentType: clients.ContentTypeJSON}, http.MethodGet),
+		},
+		{
+			"no content type header provided",
+			map[string]string{
+				clients.ContentType: ""},
+			newRequestWithHeaders(map[string]string{
+				clients.ContentType: ""}, http.MethodGet),
+		},
+		{
+			"user-agent header not propagated",
+			map[string]string{
+				"User-Agent": ""},
```

Comment:
For these negative cases could we use some uncommon header name, say `NonPropegatedHeaderName`? This way if we add `User-Agent`, `Content-MD5` in the future we will not have a series of tests which break.